### ISSUE TITLE
Flat Button does not ignore the Focus style

### ISSUE THREAD

**Issue by madmiraal** [godotengine/godot#45823]
**Godot version:**
3.2 all versions including 3.2.4.rc1

**OS/device including version:**
Ubuntu 20.10

**Issue description:**
A `Button` with the [`flat`](https://docs.godotengine.org/en/latest/classes/class_button.html#class-button-property-flat) property enabled, should ignore the decorations (styles) applied to it. However, it doesn't ignore the `Focus` style applied to it.

**Steps to reproduce:**
Create a `Button`, enable the `flat` property and apply a `Focus` `StyleBox`. Run the game and click the button to focus it, notice that the `Focus` `StyleBox` is applied. Note that the `Normal`, `Hover` and `Disabled` styles are not applied as expected.

**Minimal reproduction project:**
[FlatButtonFocusBug.zip](https://github.com/godotengine/godot/files/5943357/FlatButtonFocusBug.zip)

**Comment by Calinou** [comment]
I think this was done by design for accessibility purposes. If we remove focus feedback entirely, there's no way for keyboard-/gamepad-only users to know their current location on the GUI.

**Comment by groud** [comment]
Yep, for accessibility we still need a way to see if the button has the focus. Unless there's a cleaner way to solve this, I thinks the current situation is ok.

**Comment by madmiraal** [comment]
> we still need a way to see if the button has the focus

If so, then I would expect a simple highlight: either a thin border line or a modulated colour. What I don't expect is for all the adornments I've applied to my `Button` `Theme` to be applied. Note: In the MRP I've only changed the colour to highlight the issue, but my `Button`s have corners, borders and shadows, which are suddenly applied to a `flat` `Button` when the `Button` has focus.

It's worth noting that this doesn't just apply to `Button`s, but to everything derived from `Button`:
- CheckBox
- CheckButton
- ColorPickerButton
- MenuButton (Flat by default, but isn't affected)
- OptionButton
- ToolButton (Flat by default)

In my particular case, it's the `CheckBox` "Button" that's affected. I don't want the `Button`'s adornments to apply, so I check the `flat` property, but the `Style` is applied when it receives focus, for example, when the checkbox is ticked:
<table><tr><td>
Unfocused

![Flat Unfocused](https://user-images.githubusercontent.com/9253928/107254398-2f081c80-6a2f-11eb-89aa-ce27bf98eddf.png)

</td><td>
Focused

![Flat Focused](https://user-images.githubusercontent.com/9253928/107254492-447d4680-6a2f-11eb-8e1d-e38df7fc57cf.png)

</td></tr></table>

Likewise, I wouldn't expect all the adornments to be applied to the `Button`s and their derivatives inside, for example, the ColorPicker:
![ColorPicker2](https://user-images.githubusercontent.com/9253928/107255270-0896b100-6a30-11eb-8e52-a9028cd31d84.png)

> Unless there's a cleaner way to solve this, I thinks the current situation is ok.

I think, the right approach is to create and apply the desired style as required; not to override the `flat` (unadorned) `Button` with the adorned version just for the focused state.

**Comment by Calinou** [comment]
The Focus style is expected to draw *only* an outline, not a background. This way, it plays well with any kind of GUI element.

**Comment by jamie-pate** [comment]
I think the issue becomes the inability to separately style the focus stylebox for flat/normal buttons then?

**Comment by aragubas** [comment]
any progress? it seems to have been abandoned since September 8, 2021

**Comment by Calinou** [comment]
> any progress? it seems to have been abandoned since September 8, 2021

As explained above, this is done by design for accessibility reasons, so I don't think anything should be changed with regards to focus stylebox drawing. That said, you can apply a StyleBoxEmpty as the `focus` style override for buttons that have their Flat property enabled.

Either way, I think https://github.com/godotengine/godot-proposals/issues/2011 is a better solution to the problem described here.
